dissect/target/plugins/os/windows/defender.py

```diff
@@ -106,16 +397,142 @@ def evtx(self) -> Generator[Record, None, None]:
+                    # fully fits into the resource_id. If so, we assume that that is the matching file and break.
+                    for possible_file in subdirectory.iterdir():
+                        _, _, filename = str(possible_file).rpartition("/")
+                        if filename in entry_resource.resource_id:
+                            resourcedata_location = resourcedata_directory.joinpath(resource_id_key).joinpath(filename)
+                            break
+                    if resourcedata_location is None:
+                        self.target.log.warning(f"Could not find a ResourceData file for {entry_resource.resource_id}.")
+                        continue
+                    if resourcedata_location in recovered_files:
+                        # We already recovered this file
+                        continue
+                    fh = resourcedata_location.open()
+                    # We restore the file with the resource_id as its filename. While we could 'guess' the filename
+                    # based on the information we have from the associated quarantine entry, there is a potential that
+                    # different files have the same filename. Analysts can use the quarantine records to cross
+                    # reference.
+                    for dest_filename, dest_buf in recover_quarantined_file(fh, entry_resource.resource_id):
+                        output_filename = output_dir.joinpath(dest_filename)
+                        self.target.log.info(f"Saving {output_filename}")
+                        with open(output_filename, "wb") as output_file:
+                            output_file.write(dest_buf)
+                    fh.close()
+
+                    # Make sure we do not recover the same file multiple times if it has multiple entries
+                    recovered_files.append(resourcedata_location)
+
+    def check_compatible(self):
+        """
+        Either the Defender log folder or the quarantine folder has to exist for this plugin to be compatible.
+        """
+        return any(
+            [
+                self.target.fs.path(DEFENDER_LOG_DIR).exists(),
+                self.target.fs.path(DEFENDER_QUARANTINE_DIR).exists(),
+            ]
+        )
+
+    def get_quarantine_entries(self) -> Iterator[QuarantineEntry]:
+        """
+        For a given target, return quarantine entries of Windows Defender
+        """
+        quarantine_directory = self.target.fs.path(DEFENDER_QUARANTINE_DIR)
+        entries_directory = quarantine_directory.joinpath("entries")
+
+        if not entries_directory.is_dir():
+            return
+        for guid_path in entries_directory.iterdir():
+            if not guid_path.exists() or not guid_path.is_file():
+                continue
+            entry_fh = guid_path.open()
+            quarantine_entry = QuarantineEntry(entry_fh)
+            entry_fh.close()
```

Comment:
```suggestion
            with guid_path.open()
                quarantine_entry = QuarantineEntry(entry_fh)
```
Technically closing isn't even necessary with Dissect, but I believe this would also work (which would be the preferred way if we were to start closing)